src/main/java/io/airlift/slice/SliceUtf8.java

```diff
@@ -466,8 +466,9 @@ public static int tryGetCodePointAt(Slice utf8, int position)
 
         if (length == 2) {
             // 110x_xxxx 10xx_xxxx
-            return ((firstByte & 0b0001_1111) << 6) |
+            int codePoint = ((firstByte & 0b0001_1111) << 6) |
                     (secondByte & 0b0011_1111);
+            return codePoint < 0x80 ? -2 : codePoint;
```

Comment:
Add a comment before this line that mentions overlong encoding. It's not super clear to readers why we are doing this otherwise.